### DISCUSSION TITLE
How do I visualize the features from a checkpoint?

### DISCUSSION THREAD

**Discussion by ldlshizhu** [frgfm/torch-cam#117] [Q&A] [answered]
My model file is 'XXXX.h5' or 'XXX.pth',can I use your project to visualize the feature map?
thank you very much~

**Comment by frgfm** [comment, marked as answer]
Hi @ldlshizhu :wave: 

I'll have to make a few assumptions as I'm not sure I fully understand your question. Please provide a snippet about how you instantiate your model using the file you mention to clarify this :pray: 

So my reasoning would be as follows:
1. First, if you are actually talking about feature map visualization, this isn't the scope of this project. TorchCAM focuses on Class Activation Maps, so if a feature map is the output of a given layer, the CAM would somehow represent the spatial influence of inputs on the results.
2. Assuming we are talking about CAMs (and not feature maps), once you have instantiated your model, you can easily extract the CAM using the method of your choice as shown in https://github.com/frgfm/torch-cam#retrieving-the-class-activation-map
3. If your checkpoint is not only the `state_dict` of the model but its full serialization, 2 is still valid, you only have to recreate the model using your serialized file.

I hope this helps, let me know if you have additional questions :+1:

**Reply by frgfm** [reply]
Any update @ldlshizhu? :)

**Reply by ldlshizhu** [reply]
Hi, frgfm~, your suggestion is very helpful to me! Thank you very much for your patience, because I saw that you closed this question on github, so I didn't reply to you, so as not to disturb you again. Just now, I discovered that this issue was under 'Discussions'.hhhh
Wish you a happy life~
